scripts/deploy_test.py

```diff
@@ -0,0 +1,598 @@
+            return 'repo'
+        def mock_check_blocking_bug_issue_count(unused_repo):
+            pass
+        def mock_check_prs_for_current_release_are_released(unused_repo):
+            pass
+        def mock_check_output(unused_cmd_tokens):
+            return 'Update authors and changelog for v1.2.3'
+        config_swap = self.swap(update_configs, 'main', mock_main)
+        get_token_swap = self.swap(
+            common, 'get_personal_access_token', mock_get_personal_access_token)
+        get_org_swap = self.swap(
+            github.Github, 'get_organization', mock_get_organization)
+        get_repo_swap = self.swap(
+            github.Organization.Organization, 'get_repo', mock_get_repo)
+        bug_check_swap = self.swap(
+            common, 'check_blocking_bug_issue_count',
+            mock_check_blocking_bug_issue_count)
+        pr_check_swap = self.swap(
+            common, 'check_prs_for_current_release_are_released',
+            mock_check_prs_for_current_release_are_released)
+        out_swap = self.swap(
+            subprocess, 'check_output', mock_check_output)
+        with self.get_branch_swap, self.install_swap, self.cwd_check_swap:
+            with self.release_script_exist_swap, self.gcloud_available_swap:
+                with self.run_swap, config_swap, get_token_swap, get_org_swap:
+                    with get_repo_swap, bug_check_swap, pr_check_swap, out_swap:
+                        with args_swap, feconf_swap, self.assertRaisesRegexp(
+                            Exception,
+                            'The mailgun API key must be added before '
+                            'deployment.'):
+                            deploy.execute_deployment()
+
+    def test_missing_third_party_dir(self):
+        third_party_swap = self.swap(deploy, 'THIRD_PARTY_DIR', 'INVALID_DIR')
+        with self.get_branch_swap, self.install_swap, self.cwd_check_swap:
+            with self.release_script_exist_swap, self.gcloud_available_swap:
+                with self.run_swap, self.args_swap:
+                    with third_party_swap, self.assertRaisesRegexp(
+                        Exception,
+                        'Could not find third_party directory at INVALID_DIR. '
+                        'Please run install_third_party_libs.py prior to '
+                        'running this script.'):
+                        deploy.execute_deployment()
+
+    def test_invalid_dir_access(self):
+        def mock_getcwd():
+            return 'invalid'
+        getcwd_swap = self.swap(os, 'getcwd', mock_getcwd)
+        with self.get_branch_swap, self.install_swap, self.cwd_check_swap:
+            with self.release_script_exist_swap, self.gcloud_available_swap:
+                with self.args_swap, self.exists_swap, self.check_output_swap:
+                    with self.dir_exists_swap, self.copytree_swap, self.cd_swap:
+                        with self.run_swap, getcwd_swap:
```

Comment:
Can't you use just a one `with` and wrap the context managers inside brackets?

Reply:
Ditto in other palces.

Reply:
No, wrapping multiple swaps in bracket results in an attribute error:

```
AttributeError: __exit__
```